File: typescript/projects/Rylogic.TextAligner/src/align/align_pos.ts
import { Range } from './range';

/** */
export class AlignPos
{
	constructor(column: number, span:Range)
	{
		this.column = column;
		this.span = span;
	}

	/** The column index to align to */
	public readonly column: number;

	/** The range of characters around the align column */
	public readonly span :Range;

	/** ToString */
	public toString(): string
	{
		return `Col ${this.column} ${this.span}`;
	}
}

File: projects/apps/Rylogic.TextAligner/Shared/src/Align/AlignPos.cs
using Rylogic.Common;

namespace Rylogic.TextAligner
{
	/// <summary></summary>
	internal struct AlignPos
	{
		public AlignPos(int column, RangeI span)
		{
			Column = column;
			Span = span;
		}

		/// <summary>The column index to align to</summary>
		public readonly int Column;

		/// <summary>The range of characters around the align column</summary>
		public readonly RangeI Span;

		/// <summary></summary>
		public override string ToString() => $"Col {Column} {Span}";
	}
}
